### PR TITLE
sql: Add type column to SHOW CONNECTIONS

### PR DESCRIPTION
**Pull request by jkosh44**
Fixes: #14643

### Motivation
This PR adds a known-desirable feature.

### Checklist

- [X] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way) and therefore is tagged with a `T-protobuf` label.
- [X] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):

  - Adds type column to `SHOW CONNECTIONS`

### CODE REVIEW

File: doc/user/content/sql/show-connections.md [hunk 1]
```diff
@@ -25,8 +25,8 @@ SHOW CONNECTIONS;
 ```
 
 ```nofmt
-       name
----------------------
+       name          | type
+---------------------+------
  kafka_connection
```
Comment (benesch) [inline doc/user/content/sql/show-connections.md:30]: The data rows here are missing their types, no?

File: doc/user/content/sql/show-connections.md [hunk 2]
```diff
@@ -36,8 +36,8 @@ SHOW CONNECTIONS LIKE 'kafka%';
 ```
 
 ```nofmt
-       name
------------------
+       name      | type
+-----------------+------
```
Comment (benesch) [inline doc/user/content/sql/show-connections.md:40]: Ditto.

**Review by benesch — APPROVED** [review]

File: doc/user/content/sql/show-connections.md (hunk 1 above)
Comment (jkosh44) [reply doc/user/content/sql/show-connections.md]: Yep, just fixed it.

File: doc/user/content/sql/show-connections.md (hunk 2 above)
Comment (jkosh44) [reply doc/user/content/sql/show-connections.md:40]: Just fixed.

**Pull request merged**